src/Components/VersionHistoryPanel.test.jsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react'
+import ShareMock from '../ShareMock'
+import {render, renderHook, act, waitFor} from '@testing-library/react'
+import VersionHistoryPanel from './VersionHistoryPanel'
+import {getCommitsForBranch} from '../utils/GitHub'
+import {
+  MOCK_MODEL_PATH_GIT,
+  MOCK_REPOSITORY,
+  MOCK_COMMITS,
+} from '../utils/GitHub'
+import useStore from '../store/useStore'
+
+
+jest.mock('../utils/GitHub', () => ({
+  getCommitsForBranch: jest.fn(() => Promise.resolve([
+    MOCK_COMMITS,
+  ])),
+}))
+
+describe('VersionsHistoryPanel', () => {
+  it('fetches commits on mount', async () => {
+    const {result} = renderHook(() => useStore((state) => state))
+    await act(() => {
+      result.current.setModelPath(MOCK_MODEL_PATH_GIT)
+      result.current.setRepository(MOCK_REPOSITORY)
+    })
+    getCommitsForBranch.mockResolvedValueOnce(MOCK_COMMITS)
+    render(
+        <ShareMock>
+          <VersionHistoryPanel branch="main"/>
+        </ShareMock>,
+    )
+    await waitFor(() => {
+      expect(getCommitsForBranch).toHaveBeenCalled()
```

Comment:
I think the better testing approach here would be to remove the Jest mock of `getCommitsForBranch` altogether and instead render the component and search its text (or the document text) for the commit message.

This way you have confidence that the commit is being rendered in the timeline rather than just the function being called.